### PR TITLE
Fix patch extension for rootable collections and error handling

### PR DESCRIPTION
**Pull request by DeathByTape**

### CODE REVIEW

**Comment by swathi9488** [comment]
Looks good.

**Comment by yahoocla** [comment]
Thank you for submitting this pull request, however I do not see a valid CLA on file for you. Before we can merge this request please visit https://yahoocla.herokuapp.com/ and agree to the terms. Thanks! :smile:

**Comment by yahoocla** [comment]
CLA is valid!

File: elide-core/src/main/java/com/yahoo/elide/core/PersistentResource.java [hunk 1]
```diff
@@ -255,7 +255,10 @@ public boolean matchesId(String checkId) {
         }
 
         PersistentResource<T> resource = new PersistentResource<>(obj, requestScope);
-        resource.checkFieldAwarePermissions(ReadPermission.class);
+        // No need to have read access for a newly created object
+        if (!requestScope.getNewResources().contains(resource)) {
+            resource.checkFieldAwarePermissions(ReadPermission.class);
```
Comment (aklish) [inline elide-core/src/main/java/com/yahoo/elide/core/PersistentResource.java]: Why would loadRecord be called for a newly created resource?

**Comment by aklish** [comment]
:+1:

**Pull request merged**